core/unit_test/tstHDF5ParticleOutput.hpp

```diff
@@ -0,0 +1,187 @@
+
+    Cabana::Experimental::HDF5ParticleOutput::HDF5Config h5_config;
+
+    h5_config.collective = true;
+
+    // Write a time step to file.
+    double time_read;
+    double time = 7.64;
+    double step = 892;
+    Cabana::Experimental::HDF5ParticleOutput::writeTimeStep(
+        h5_config, "particles", MPI_COMM_WORLD, step, time, coords.size(),
+        coords, ids, matrix, vec );
+
+    // Make an empty copy to read into.
+    auto aosoa_read = Cabana::create_mirror_view( Kokkos::HostSpace(), aosoa );
+    auto coords_read = Cabana::slice<0>( aosoa_read, "coords" );
+    auto vec_read = Cabana::slice<1>( aosoa_read, "vec" );
+    auto matrix_read = Cabana::slice<2>( aosoa_read, "matrix" );
+    auto ids_read = Cabana::slice<3>( aosoa_read, "ids" );
+
+    // Read the data back in and compare.
+    Cabana::Experimental::HDF5ParticleOutput::readTimeStep(
+        h5_config, "particles", MPI_COMM_WORLD, step, coords.size(),
+        coords.label(), time_read, coords_read );
+    checkVector( coords_mirror, coords_read );
+    EXPECT_DOUBLE_EQ( time, time_read );
+
+    Cabana::Experimental::HDF5ParticleOutput::readTimeStep(
+        h5_config, "particles", MPI_COMM_WORLD, step, coords.size(),
+        ids.label(), time_read, ids_read );
+    checkScalar( ids_mirror, ids_read );
+    EXPECT_DOUBLE_EQ( time, time_read );
+
+    Cabana::Experimental::HDF5ParticleOutput::readTimeStep(
+        h5_config, "particles", MPI_COMM_WORLD, step, coords.size(),
+        vec.label(), time_read, vec_read );
+    checkVector( vec_mirror, vec_read );
+    EXPECT_DOUBLE_EQ( time, time_read );
+
+    Cabana::Experimental::HDF5ParticleOutput::readTimeStep(
+        h5_config, "particles", MPI_COMM_WORLD, step, coords.size(),
+        matrix.label(), time_read, matrix_read );
+    checkMatrix( matrix_mirror, matrix_read );
+    EXPECT_DOUBLE_EQ( time, time_read );
+
+    // Move the particles and write again.
+    double time_step_size = 0.32;
+    time += time_step_size;
+    ++step;
+    for ( int p = 0; p < num_particle; ++p )
+        for ( int d = 0; d < 3; ++d )
+            coords_mirror( p, d ) += 1.32;
+    Cabana::deep_copy( coords, coords_mirror );
+    Cabana::Experimental::HDF5ParticleOutput::writeTimeStep(
+        h5_config, "particles", MPI_COMM_WORLD, step, time, coords.size(),
+        coords, ids, matrix, vec );
+
+    // Read the data back in and compare.
+    Cabana::Experimental::HDF5ParticleOutput::readTimeStep(
+        h5_config, "particles", MPI_COMM_WORLD, step, coords_read.size(),
+        coords.label(), time_read, coords_read );
+    checkVector( coords_mirror, coords_read );
+    EXPECT_DOUBLE_EQ( time, time_read );
+}
+
+//---------------------------------------------------------------------------//
+// RUN TESTS
+//---------------------------------------------------------------------------//
+TEST( TEST_CATEGORY, write_read_test ) { writeReadTest(); }
+
+// FIXME: Add test with ghosted particles
```

Comment:
Huh?

Reply:
Removed - seemed like a good extra test at the time, but now it seems out of scope